pkg/controller/pod_test.go

```diff
@@ -353,6 +353,268 @@ var _ = Describe("Dynamic Attachment controller", func() {
 				})
 			})
 
+			When("a wrong attachment is added to the pod's network annotations with a following correct attachement", func() {
+				JustBeforeEach(func() {
+					pod = updatePodSpec(pod)
+					netSelectionElements := append(generateNetworkSelectionElements(namespace, networkName),
+						[]nad.NetworkSelectionElement{
+							{
+								Name:             networkToAdd,
+								Namespace:        namespace,
+								InterfaceRequest: "net-non-existing",
+							},
+							{
+								Name:             networkToAdd,
+								Namespace:        namespace,
+								InterfaceRequest: "net1",
+							},
+						}...,
+					)
+					serelizedNetSelectionElements, _ := json.Marshal(netSelectionElements)
+					pod.Annotations[nad.NetworkAttachmentAnnot] = string(serelizedNetSelectionElements)
+					var err error
+					_, err = k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						pod,
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("an `FailedAddingInterface` event is seens, no `AddedInterface` event is seen and no changes the status", func() {
+					expectedEventPayload := fmt.Sprintf(
+						"Warning FailedAddingInterface pod [%s]: failed adding interface %s to network: %s",
+						annotations.NamespacedName(namespace, podName),
+						"net-non-existing",
+						networkToAdd,
+					)
+					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
+
+					// reconciliation requeued without adding the next interface (net1).
+					expectedEventPayload = fmt.Sprintf(
+						"Warning FailedAddingInterface pod [%s]: failed adding interface %s to network: %s",
+						annotations.NamespacedName(namespace, podName),
+						"net-non-existing",
+						networkToAdd,
+					)
+					Eventually(<-eventRecorder.Events).Should(Equal(expectedEventPayload))
+
+					// This is not in a separate "It" since there is no change and it should wait for the events
+					// No pod network-status is added since the first one failed.
+					Eventually(func() ([]nad.NetworkStatus, error) {
```

Comment:
shouldn't this be a consistently ?

Reply:
Right, I fixed it with a 1 second timeout and 100 ms polling time